lua/nvim-tree/renderer/init.lua

```diff
@@ -17,11 +17,14 @@ local M = {
 
 local namespace_id = api.nvim_create_namespace "NvimTreeHighlights"
 
-local function _draw(bufnr, lines, hl)
+local function _draw(bufnr, lines, hl, signs)
   api.nvim_buf_set_option(bufnr, "modifiable", true)
   api.nvim_buf_set_lines(bufnr, 0, -1, false, lines)
   M.render_hl(bufnr, hl)
   api.nvim_buf_set_option(bufnr, "modifiable", false)
+  for _, sign in pairs(signs) do
+    vim.fn.sign_place(1, git.SIGN_GROUP, sign.sign, bufnr, { lnum = sign.lnum, priority = 1 })
```

Comment:
not sure about the ID, i do not understand what it's used for. I've set it just like the one in diagnostics.

Reply:
sign-identifier seems to be used for mark like functionality, which we don't really need.

You could just call sign_place with 0 and have vim allocate a unique one.